### PR TITLE
test: expand test coverage across all packages

### PR DESCRIPTION
**Pull request by cboone**
Add comprehensive test coverage for command execution, config loading,
container operations, and output formatting. Coverage improvements:
- cmd: 18.8% → 78.3% (+59.5%)
- internal/config: 66.3% → 92.5% (+26.2%)
- internal/container: 66.7% → 78.9% (+12.2%)
- internal/output: 91.6% maintained

Key additions:
- Test infrastructure with MockRunner for cmd package
- Command execution tests for all 17 commands
- Helper function tests (prepareWorkspace, handleBuildFlags, applyFlags)
- Load/findConfigFile/findDataDir tests for config
- generateContainerfile/copyDir/mount tests for container
- Edge case and format string tests for output

### CODE REVIEW

File: internal/config/config_test.go [hunk 1]
```diff
@@ -254,5 +254,207 @@ window_prefix: "test-"
...
+			gomega.Expect(cfg.Memory).To(gomega.Equal("16g"))
+			gomega.Expect(cfg.ConfigFile).To(gomega.Equal(configFile))
+		})
+
+		It("uses defaults when no config file exists", func() {
+			cfg, err := Load("")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(cfg.Memory).To(gomega.Equal(DefaultMemory))
+			gomega.Expect(cfg.CPUs).To(gomega.Equal(DefaultCPUs))
+		})
```
[the remark this answers is not in the document]
Comment (cboone) [reply internal/config/config_test.go]: This is a macOS-only tool designed to run on developer machines. Sensitivity to ambient config is acceptable for integration-style testing in this context. Added clarification to copilot-instructions.md under "Test environment assumptions."

File: cmd/cmd_test.go [hunk 2]
```diff
@@ -292,4 +409,912 @@ var _ = Describe("Commands", func() {
...
+
+		AfterEach(func() {
+			restore()
+		})
+
+		It("calls RunClaude with workspace", func() {
+			err := claudeCmd.RunE(claudeCmd, []string{})
+			// RunClaude uses RunInteractive internally, verify it was attempted
+			_ = err // May fail due to missing image, but we verify the call path was exercised
+			Expect(len(mock.RunInteractiveCalls)).To(BeNumerically(">=", 0))
```
[the remark this answers is not in the document]
Comment (cboone) [reply cmd/cmd_test.go]: Fixed by removing the tautological assertion. The test now verifies the code path is exercised without panicking, with a comment explaining that RunInteractive may not be called if the command fails early due to workspace resolution in the test environment.

File: internal/container/manager_test.go [hunk 3]
```diff
@@ -657,3 +657,248 @@ var _ = Describe("Container", func() {
...
+		var (
+			cfg     *config.Config
+			mock    *MockRunner
+			manager *Manager
+		)
+
+		BeforeEach(func() {
+			cfg = config.Default()
+			mock = NewMockRunner()
+			manager = &Manager{Config: cfg, Runner: mock}
```
[the remark this answers is not in the document]
Comment (cboone) [reply internal/container/manager_test.go]: Fixed in commit 5f2743c. Now using `NewTestManager(cfg, mock)` which properly initializes the `System` field.

File: internal/container/testing.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+// SetPlatformMocks sets the platform check functions for testing from external packages.
+// Returns a restore function that should be called to reset the original values.
+func SetPlatformMocks(osName, arch string, lookupErr error) func() {
+	return setPlatformMocksImpl(osName, arch, lookupErr)
+}
+
+// SetKernelMock sets the kernel directory check function for testing from external packages.
+// Returns a restore function that should be called to reset the original value.
+func SetKernelMock(empty bool, err error) func() {
```
[the remark this answers is not in the document]
Comment (cboone) [reply internal/container/testing.go]: This is a non-issue for this project. The Makefile runs `ginkgo -v -r -race -cover` which executes packages **sequentially** by default (no `-p` or `--procs` flag). Parallel package execution would require explicit `-p` which is not used. Additionally, all tests use BeforeEach/AfterEach patterns that properly restore global state after each spec. This pattern is acceptable for a macOS-only developer tool.
